scripts/query_testlists

```diff
@@ -61,15 +67,38 @@ def parse_command_line(args, description):
 
     args = CIME.utils.parse_args_and_handle_standard_logging_options(args, parser)
 
-    expect(not(args.count and args.list_type),
-           "Cannot specify both --count and --list arguments.")
+    _check_argument_compatibility(args)
 
     if args.list_type:
         _process_list_type(args)
 
     return args
 
 ###############################################################################
+def _check_argument_compatibility(args):
+###############################################################################
+    """Ensures there are no incompatible arguments
+
+    If incompatible arguments are found, aborts with a helpful error
+    message.
+    """
+
+    expect(not(args.count and args.list_type),
+           "Cannot specify both --count and --list arguments.")
+
+    if args.count:
```

Comment:
This might be cleaner to achieve with ArgParse's mutually exclusive group (see case.build).

Reply:
That's a useful-looking feature. But I'm struggling with how to do that here. I see how I can replace this with a mutually exclusive group:

```python
    expect(not(args.count and args.list_type),
           "Cannot specify both --count and --list arguments.")
```

But I can't see how to express the rest of the logic with that mechanism: Although it's not allowed to combine either `--show-options` or `--define-testtypes` with either `--list` or `--count`, it _is_ allowed to combine `--show-options` with `--define-testtypes`. Can you see how to express this with a mutually exclusive group?

Reply:
OK, if it's not obvious then don't worry about it.